src/app/store/effects.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import {
+  getLogin,
+  getLoginComplete,
+  getLogout,
+  getLogoutComplete,
+  getRegister,
+  getRegisterComplete,
+} from './actions';
+import { AuthService } from '../services/auth.service';
+import { catchError, EMPTY, map, retry, switchMap } from 'rxjs';
+
+@Injectable()
+export class UserEffects {
+  getlogin$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(getLogin.type),
+      switchMap((action: { email: string; password: string }) =>
+        this.authService.login(action.email, action.password).pipe(
+          map(({ name, id }) => getLoginComplete({ name, id })),
+          retry(1),
+          catchError((error) => {
+            console.error(
```

Comment:
I would suggest for this since the error is for the user, maybe use a notification service/ an alert since an average user will not open the console to check whats happening.

Reply:
I am planning on adding a lot more visible feedback once I work on the UI again, I definitely want to add the notifications.